File: src/main/java/org/zbus/rpc/RpcConfig.java
/**
 * The MIT License (MIT)
 * Copyright (c) 2009-2015 HONG LEIMING
 * 
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package org.zbus.rpc;

import org.zbus.mq.MqConfig;

public class RpcConfig extends MqConfig {
	public static final String DEFAULT_ENCODING = "UTF-8";  
	
	protected String module = ""; 
	protected int timeout = 10000;
	protected String encoding = DEFAULT_ENCODING;
	
	public String getModule() {
		return module;
	}
	public void setModule(String module) {
		this.module = module;
	}
	public int getTimeout() {
		return timeout;
	}
	public void setTimeout(int timeout) {
		this.timeout = timeout;
	}
	public String getEncoding() {
		return encoding;
	}
	public void setEncoding(String encoding) {
		this.encoding = encoding;
	}
	
}

File: NET/Zbus/Rpc/RpcConfig.cs

using Zbus.Mq;

namespace Zbus.RPC
{
   public class RpcConfig : MqConfig
   {
      public static readonly string DEFAULT_ENCODING = "UTF-8";
      private string module = "";
      private int timeout = 10000;
      private string encoding = DEFAULT_ENCODING;

      public string Module
      {
         get { return module; }
         set { module = value; }
      }

      public int Timeout
      {
         get { return timeout; }
         set { timeout = value; }
      }

      public string Encoding
      {
         get { return encoding; }
         set { encoding = value; }
      }
   }
}
